### ISSUE TITLE
Prometheus queries with over 7.7k characters fail

### ISSUE THREAD

**Issue by andrewmcgilvray** [grafana/grafana#65311]
**What happened**:
Prometheus queries fail if their length is somewhere between 7608 and 7695 since the move to querydata a.k.a. streaming parser in 9.3. With Grafana 9.2.15 I can perform prometheus queries of over 12k successfully.

9.3.x fails with an error.
<img width="593" alt="image" src="https://user-images.githubusercontent.com/1414394/227409613-d5c333a0-a7cc-4417-a3a6-ecc599511008.png">

9.4.x fails silently.
<img width="469" alt="image" src="https://user-images.githubusercontent.com/1414394/227408579-44c7ee3a-c700-4185-9382-5784e31cace3.png">

**What you expected to happen**:
The query to return with results. If there is an error it should be displayed.

**How to reproduce it (as minimally and precisely as possible)**:
Create a Prometheus query with over 7700 characters in it.

**Anything else we need to know?**:

**Environment**:
- Grafana version: 9.3 and above
- Data source type & version: prometheus 2.39.1
- OS Grafana is installed on: ubuntu grafana docker container
- User OS & Browser: macos and chrome
- Grafana plugins:
- Others:

**Comment by gtk-grafana** [comment]
@andrewmcgilvray I can't reproduce this on modern Grafana, here's a ~10k prom query that is working (not throwing 500) on play.grafana.org.

It looks like this was fixed in 9.3.2 with https://github.com/grafana/grafana/issues/60242.
Closing as I cannot reproduce, but please re-open if you are still running into this!

```
https://play.grafana.org/explore?orgId=1&left=%7B%22datasource%22:%22grafanacloud-prom%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22datasource%22:%7B%22type%22:%22prometheus%22,%22uid%22:%22grafanacloud-prom%22%7D,%22editorMode%22:%22code%22,%22expr%22:%22node_arp_entries%7Bagent_hostname%3D%5C%22Lorem%20ipsum%20dolor%20sit%20amet,%20consectetuer%20adipiscing%20elit.%20Aenean%20commodo%20ligula%20eget%20dolor.%20Aenean%20massa.%20Cum%20sociis%20natoque%20penatibus%20et%20magnis%20dis%20parturient%20montes,%20nascetur%20ridiculus%20mus.%20Donec%20quam%20felis,%20ultricies%20nec,%20pellentesque%20eu,%20pretium%20quis,%20sem.%20Nulla%20consequat%20massa%20quis%20enim.%20Donec%20pede%20justo,%20fringilla%20vel,%20aliquet%20nec,%20vulputate%20eget,%20arcu.%20In%20enim%20justo,%20rhoncus%20ut,%20imperdiet%20a,%20venenatis%20vitae,%20justo.%20Nullam%20dictum%20felis%20eu%20pede%20mollis%20pretium.%20Integer%20tincidunt.%20Cras%20dapibus.%20Vivamus%20elementum%20semper%20nisi.%20Aenean%20vulputate%20eleifend%20tellus.%20Aenean%20leo%20ligula,%20porttitor%20eu,%20consequat%20vitae,%20eleifend%20ac,%20enim.%20Aliquam%20lorem%20ante,%20dapibus%20in,%20viverra%20quis,%20feugiat%20a,%20tellus.%20Phasellus%20viverra%20nulla%20ut%20metus%20varius%20laoreet.%20Quisque%20rutrum.%20Aenean%20imperdiet.%20Etiam%20ultricies%20nisi%20vel%20augue.%20Curabitur%20ullamcorper%20ultricies%20nisi.%20Nam%20eget%20dui.%20Etiam%20rhoncus.%20Maecenas%20tempus,%20tellus%20eget%20condimentum%20rhoncus,%20sem%20quam%20semper%20libero,%20sit%20amet%20adipiscing%20sem%20neque%20sed%20ipsum.%20Nam%20quam%20nunc,%20blandit%20vel,%20luctus%20pulvinar,%20hendrerit%20id,%20lorem.%20Maecenas%20nec%20odio%20et%20ante%20tincidunt%20tempus.%20Donec%20vitae%20sapien%20ut%20libero%20venenatis%20faucibus.%20Nullam%20quis%20ante.%20Etiam%20sit%20amet%20orci%20eget%20eros%20faucibus%20tincidunt.%20Duis%20leo.%20Sed%20fringilla%20mauris%20sit%20amet%20nibh.%20Donec%20sodales%20sagittis%20magna.%20Sed%20consequat,%20leo%20eget%20bibendum%20sodales,%20augue%20velit%20cursus%20nunc,%20quis%20gravida%20magna%20mi%20a%20libero.%20Fusce%20vulputate%20eleifend%20sapien.%20Vestibulum%20purus%20quam,%20scelerisque%20ut,%20mollis%20sed,%20nonummy%20id,%20metus.%20Nullam%20accumsan%20lorem%20in%20dui.%20Cras%20ultricies%20mi%20eu%20turpis%20hendrerit%20fringilla.%20Vestibulum%20ante%20ipsum%20primis%20in%20faucibus%20orci%20luctus%20et%20ultrices%20posuere%20cubilia%20Curae;%20In%20ac%20dui%20quis%20mi%20consectetuer%20lacinia.%20Nam%20pretium%20turpis%20et%20arcu.%20Duis%20arcu%20tortor,%20suscipit%20eget,%20imperdiet%20nec,%20imperdiet%20iaculis,%20ipsum.%20Sed%20aliquam%20ultrices%20mauris.%20Integer%20ante%20arcu,%20accumsan%20a,%20consectetuer%20eget,%20posuere%20ut,%20mauris.%20Praesent%20adipiscing.%20Phasellus%20ullamcorper%20ipsum%20rutrum%20nunc.%20Nunc%20nonummy%20metus.%20Vestibulum%20volutpat%20pretium%20libero.%20Cras%20id%20dui.%20Aenean%20ut%20eros%20et%20nisl%20sagittis%20vestibulum.%20Nullam%20nulla%20eros,%20ultricies%20sit%20amet,%20nonummy%20id,%20imperdiet%20feugiat,%20pede.%20Sed%20lectus.%20Donec%20mollis%20hendrerit%20risus.%20Phasellus%20nec%20sem%20in%20justo%20pellentesque%20facilisis.%20Etiam%20imperdiet%20imperdiet%20orci.%20Nunc%20nec%20neque.%20Phasellus%20leo%20dolor,%20tempus%20non,%20auctor%20et,%20hendrerit%20quis,%20nisi.%20Curabitur%20ligula%20sapien,%20tincidunt%20non,%20euismod%20vitae,%20posuere%20imperdiet,%20leo.%20Maecenas%20malesuada.%20Praesent%20congue%20erat%20at%20massa.%20Sed%20cursus%20turpis%20vitae%20tortor.%20Donec%20posuere%20vulputate%20arcu.%20Phasellus%20accumsan%20cursus%20velit.%20Vestibulum%20ante%20ipsum%20primis%20in%20faucibus%20orci%20luctus%20et%20ultrices%20posuere%20cubilia%20Curae;%20Sed%20aliquam,%20nisi%20quis%20porttitor%20congue,%20elit%20erat%20euismod%20orci,%20ac%20placerat%20dolor%20lectus%20quis%20orci.%20Phasellus%20consectetuer%20vestibulum%20elit.%20Aenean%20tellus%20metus,%20bibendum%20sed,%20posuere%20ac,%20mattis%20non,%20nunc.%20Vestibulum%20fringilla%20pede%20sit%20amet%20augue.%20In%20turpis.%20Pellentesque%20posuere.%20Praesent%20turpis.%20Aenean%20posuere,%20tortor%20sed%20cursus%20feugiat,%20nunc%20augue%20blandit%20nunc,%20eu%20sollicitudin%20urna%20dolor%20sagittis%20lacus.%20Donec%20elit%20libero,%20sodales%20nec,%20volutpat%20a,%20suscipit%20non,%20turpis.%20Nullam%20sagittis.%20Suspendisse%20pulvinar,%20augue%20ac%20venenatis%20condimentum,%20sem%20libero%20volutpat%20nibh,%20nec%20pellentesque%20velit%20pede%20quis%20nunc.%20Vestibulum%20ante%20ipsum%20primis%20in%20faucibus%20orci%20luctus%20et%20ultrices%20posuere%20cubilia%20Curae;%20Fusce%20id%20purus.%20Ut%20varius%20tincidunt%20libero.%20Phasellus%20dolor.%20Maecenas%20vestibulum%20mollis%20diam.%20Pellentesque%20ut%20neque.%20Pellentesque%20habitant%20morbi%20tristique%20senectus%20et%20netus%20et%20malesuada%20fames%20ac%20turpis%20egestas.%20In%20dui%20magna,%20posuere%20eget,%20vestibulum%20et,%20tempor%20auctor,%20justo.%20In%20ac%20felis%20quis%20tortor%20malesuada%20pretium.%20Pellentesque%20auctor%20neque%20nec%20urna.%20Proin%20sapien%20ipsum,%20porta%20a,%20auctor%20quis,%20euismod%20ut,%20mi.%20Aenean%20viverra%20rhoncus%20pede.%20Pellentesque%20habitant%20morbi%20tristique%20senectus%20et%20netus%20et%20malesuada%20fames%20ac%20turpis%20egestas.%20Ut%20non%20enim%20eleifend%20felis%20pretium%20feugiat.%20Vivamus%20quis%20mi.%20Phasellus%20a%20est.%20Phasellus%20magna.%20In%20hac%20habitasse%20platea%20dictumst.%20Curabitur%20at%20lacus%20ac%20velit%20ornare%20lobortis.%20Curabitur%20a%20felis%20in%20nunc%20fringilla%20tristique.%20Morbi%20mattis%20ullamcorper%20velit.%20Phasellus%20gravida%20semper%20nisi.%20Nullam%20vel%20sem.%20Pellentesque%20libero%20tortor,%20tincidunt%20et,%20tincidunt%20eget,%20semper%20nec,%20quam.%20Sed%20hendrerit.%20Morbi%20ac%20felis.%20Nunc%20egestas,%20augue%20at%20pellentesque%20laoreet,%20felis%20eros%20vehicula%20leo,%20at%20malesuada%20velit%20leo%20quis%20pede.%20Donec%20interdum,%20metus%20et%20hendrerit%20aliquet,%20dolor%20diam%20sagittis%20ligula,%20eget%20egestas%20libero%20turpis%20vel%20mi.%20Nunc%20nulla.%20Fusce%20risus%20nisl,%20viverra%20et,%20tempor%20et,%20pretium%20in,%20sapien.%20Donec%20venenatis%20vulputate%20lorem.%20Morbi%20nec%20metus.%20Phasellus%20blandit%20leo%20ut%20odio.%20Maecenas%20ullamcorper,%20dui%20et%20placerat%20feugiat,%20eros%20pede%20varius%20nisi,%20condimentum%20viverra%20felis%20nunc%20et%20lorem.%20Sed%20magna%20purus,%20fermentum%20eu,%20tincidunt%20eu,%20varius%20ut,%20felis.%20In%20auctor%20lobortis%20lacus.%20Quisque%20libero%20metus,%20condimentum%20nec,%20tempor%20a,%20commodo%20mollis,%20magna.%20Vestibulum%20ullamcorper%20mauris%20at%20ligula.%20Fusce%20fermentum.%20Nullam%20cursus%20lacinia%20erat.%20Praesent%20blandit%20laoreet%20nibh.%20Fusce%20convallis%20metus%20id%20felis%20luctus%20adipiscing.%20Pellentesque%20egestas,%20neque%20sit%20amet%20convallis%20pulvinar,%20justo%20nulla%20eleifend%20augue,%20ac%20auctor%20orci%20leo%20non%20est.%20Quisque%20id%20mi.%20Ut%20tincidunt%20tincidunt%20erat.%20Etiam%20feugiat%20lorem%20non%20metus.%20Vestibulum%20dapibus%20nunc%20ac%20augue.%20Curabitur%20vestibulum%20aliquam%20leo.%20Praesent%20egestas%20neque%20eu%20enim.%20In%20hac%20habitasse%20platea%20dictumst.%20Fusce%20a%20quam.%20Etiam%20ut%20purus%20mattis%20mauris%20sodales%20aliquam.%20Curabitur%20nisi.%20Quisque%20malesuada%20placerat%20nisl.%20Nam%20ipsum%20risus,%20rutrum%20vitae,%20vestibulum%20eu,%20molestie%20vel,%20lacus.%20Sed%20augue%20ipsum,%20egestas%20nec,%20vestibulum%20et,%20malesuada%20adipiscing,%20dui.%20Vestibulum%20facilisis,%20purus%20nec%20pulvinar%20iaculis,%20ligula%20mi%20congue%20nunc,%20vitae%20euismod%20ligula%20urna%20in%20dolor.%20Mauris%20sollicitudin%20fermentum%20libero.%20Praesent%20nonummy%20mi%20in%20odio.%20Nunc%20interdum%20lacus%20sit%20amet%20orci.%20Vestibulum%20rutrum,%20mi%20nec%20elementum%20vehicula,%20eros%20quam%20gravida%20nisl,%20id%20fringilla%20neque%20ante%20vel%20mi.%20Morbi%20mollis%20tellus%20ac%20sapien.%20Phasellus%20volutpat,%20metus%20eget%20egestas%20mollis,%20lacus%20lacus%20blandit%20dui,%20id%20egestas%20quam%20mauris%20ut%20lacus.%20Fusce%20vel%20dui.%20Sed%20in%20libero%20ut%20nibh%20placerat%20accumsan.%20Proin%20faucibus%20arcu%20quis%20ante.%20In%20consectetuer%20turpis%20ut%20velit.%20Nulla%20sit%20amet%20est.%20Praesent%20metus%20tellus,%20elementum%20eu,%20semper%20a,%20adipiscing%20nec,%20purus.%20Cras%20risus%20ipsum,%20faucibus%20ut,%20ullamcorper%20id,%20varius%20ac,%20leo.%20Suspendisse%20feugiat.%20Suspendisse%20enim%20turpis,%20dictum%20sed,%20iaculis%20a,%20condimentum%20nec,%20nisi.%20Praesent%20nec%20nisl%20a%20purus%20blandit%20viverra.%20Praesent%20ac%20massa%20at%20ligula%20laoreet%20iaculis.%20Nulla%20neque%20dolor,%20sagittis%20eget,%20iaculis%20quis,%20molestie%20non,%20velit.%20Mauris%20turpis%20nunc,%20blandit%20et,%20volutpat%20molestie,%20porta%20ut,%20ligula.%20Fusce%20pharetra%20convallis%20urna.%20Quisque%20ut%20nisi.%20Donec%20mi%20odio,%20faucibus%20at,%20scelerisque%20quis,%20convallis%20in,%20nisi.%20Suspendisse%20non%20nisl%20sit%20amet%20velit%20hendrerit%20rutrum.%20Ut%20leo.%20Ut%20a%20nisl%20id%20ante%20tempus%20hendrerit.%20Proin%20pretium,%20leo%20ac%20pellentesque%20mollis,%20felis%20nunc%20ultrices%20eros,%20sed%20gravida%20augue%20augue%20mollis%20justo.%20Suspendisse%20eu%20ligula.%20Nulla%20facilisi.%20Donec%20id%20justo.%20Praesent%20porttitor,%20nulla%20vitae%20posuere%20iaculis,%20arcu%20nisl%20dignissim%20dolor,%20a%20pretium%20mi%20sem%20ut%20ipsum.%20Curabitur%20suscipit%20suscipit%20tellus.%20Praesent%20vestibulum%20dapibus%20nibh.%20Etiam%20iaculis%20nunc%20ac%20metus.%20Ut%20id%20nisl%20quis%20enim%20dignissim%20sagittis.%20Etiam%20sollicitudin,%20ipsum%20eu%20pulvinar%20rutrum,%20tellus%20ipsum%20laoreet%20sapien,%20quis%20venenatis%20ante%20odio%20sit%20amet%20eros.%20Proin%20magna.%20Duis%20vel%20nibh%20at%20velit%20scelerisque%20suscipit.%20Curabitur%20turpis.%20Vestibulum%20suscipit%20nulla%20quis%20orci.%20Fusce%20ac%20felis%20sit%20amet%20ligula%20pharetra%20condimentum.%20Maecenas%20egestas%20arcu%20quis%20ligula%20mattis%20placerat.%20Duis%20lobortis%20massa%20imperdiet%20quam.%20Suspendisse%20potenti.%20Pellentesque%20commodo%20eros%20a%20enim.%20Vestibulum%20turpis%20sem,%20aliquet%20eget,%20lobortis%20pellentesque,%20rutrum%20eu,%20nisl.%20Sed%20libero.%20Aliquam%20erat%20volutpat.%20Etiam%20vitae%20tortor.%20Morbi%20vestibulum%20volutpat%20enim.%20Aliquam%20eu%20nunc.%20Nunc%20sed%20turpis.%20Sed%20mollis,%20eros%20et%20ultrices%20tempus,%20mauris%20ipsum%20aliquam%20libero,%20non%20adipiscing%20dolor%20urna%20a%20orci.%20Nulla%20porta%20dolor.%20Class%20aptent%20taciti%20sociosqu%20ad%20litora%20torquent%20per%20conubia%20nostra,%20per%20inceptos%20hymenaeos.%20Pellentesque%20dapibus%20hendrerit%20tortor.%20Praesent%20egestas%20tristique%20nibh.%20Sed%20a%20libero.%20Cras%20varius.%20Donec%20vitae%20orci%20sed%20dolor%20rutrum%20auctor.%20Fusce%20egestas%20elit%20eget%20lorem.%20Suspendisse%20nisl%20elit,%20rhoncus%20eget,%20elementum%20ac,%20condimentum%20eget,%20diam.%20Nam%20at%20tortor%20in%20tellus%20interdum%20sagittis.%20Aliquam%20lobortis.%20Donec%20orci%20lectus,%20aliquam%20ut,%20faucibus%20non,%20euismod%20id,%20nulla.%20Curabitur%20blandit%20mollis%20lacus.%20Nam%20adipiscing.%20Vestibulum%20eu%20odio.%20Vivamus%20laoreet.%20Nullam%20tincidunt%20adipiscing%20enim.%20Phasellus%20tempus.%20Proin%20viverra,%20ligula%20sit%20amet%20ultrices%20semper,%20ligula%20arcu%20tristique%20sapien,%20a%20accumsan%20nisi%20mauris%20ac%20eros.%20Fusce%20neque.%20Suspendisse%20faucibus,%20nunc%20et%20pellentesque%20egestas,%20lacus%20ante%20convallis%20tellus,%20vitae%20iaculis%20lacus%20elit%20id%20tortor.%20Vivamus%20aliquet%20elit%20ac%20nisl.%20Fusce%20fermentum%20odio%20nec%20arcu.%20Vivamus%20euismod%20mauris.%20In%20ut%20quam%20vitae%20odio%20lacinia%20tincidunt.%20Praesent%20ut%20ligula%20non%20mi%20varius%20sagittis.%20Cras%20sagittis.%20Praesent%20ac%20sem%20eget%20est%20egestas%20volutpat.%20Vivamus%20consectetuer%20hendrerit%20lacus.%20Cras%20non%20dolor.%20Vivamus%20in%20erat%20ut%20urna%20cursus%20vestibulum.%20Fusce%20commodo%20aliquam%20arcu.%20Nam%20commodo%20suscipit%20quam.%20Quisque%20id%20odio.%20Praesent%20venenatis%20metus%20at%20tortor%20pulvinar%20varius.%20Lorem%20ipsum%20dolor%20sit%20amet,%20consectetuer%20adipiscing%20elit.%20Aenean%20commodo%20ligula%20eget%20dolor.%20Aenean%20massa.%20Cum%20sociis%20natoque%20penatibus%20et%20magnis%20dis%20parturient%20montes,%20nascetur%20ridiculus%20mus.%20Donec%20quam%20felis,%20ultricies%20nec,%20pellentesque%20eu,%20pretium%20quis,%20sem.%20Nulla%20consequat%20massa%20quis%20enim.%20Donec%20pede%20justo,%20fringilla%20vel,%20aliquet%20nec,%20vulputate%20eget,%20arcu.%20In%20enim%20justo,%20rhoncus%20ut,%20imperdiet%20a,%20venenatis%20vitae,%20justo.%20Nullam%20dictum%20felis%20eu%20pede%20mollis%20pretium.%20Integer%20tincidunt.%20Cras%20dapibus.%20Vivamus%20elementum%20semper%20nisi.%20Aenean%20vulputate%20eleifend%20tellus.%20Aenean%20leo%20ligula,%20porttitor%20eu,%20consequat%20vita%5C%22%7D%22,%22legendFormat%22:%22__auto%22,%22range%22:true,%22instant%22:true%7D%5D,%22range%22:%7B%22from%22:%22now-1h%22,%22to%22:%22now%22%7D%7D
```

**Comment by andrewmcgilvray** [comment]
I believe I found the issue.
I did not have a Prometheus Type and Prometheus Version set. Once I set that all of my long queries worked!
![image](https://github.com/grafana/grafana/assets/1414394/650da626-8050-4cb5-aa33-74eba308767a)
